controller/appcontroller.go

```diff
@@ -2241,8 +2244,11 @@ func (ctrl *ApplicationController) shouldSelfHeal(app *appv1.Application, alread
 		return true, time.Duration(0)
 	}
 
-	// Reset counter if the prior sync was successful OR if the revision has changed
-	if !alreadyAttempted || app.Status.Sync.Status == appv1.SyncStatusCodeSynced {
+	timeSinceOperation := time.Since(app.Status.OperationState.FinishedAt.Time)
+
+	// Reset counter if the prior sync was successful and the cooldown period is over OR if the revision has changed
+	if !alreadyAttempted || (len(app.Status.History) > 0 && timeSinceOperation >= ctrl.selfHealBackoffCooldown && app.Status.Sync.Status == appv1.SyncStatusCodeSynced) {
+		log.Printf("Resetting self-heal attempts count for %s/%s", app.Namespace, app.Name)
```

Comment:
I'd move down this block to the `else` on L2263 to make it more clear that this is only relevant for the exp backoff logic (which the if below is not a part of)

Reply:
Is there a need to also check for the app health status (probably not given that this can happen even for health apps)?